### PR TITLE
Feat/published new screen integration

### PR DESCRIPTION
**Pull request by kjawadDeveloper2**
1) Solved all the design related issues in it . 
2) Added the publish screen. 
3) Redirecting to the IPFS web

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -65,7 +65,14 @@ android {
         release {
             // TODO: Add your own signing config for the release build.
             // Signing with the debug keys for now, so `flutter run --release` works.
+
+            shrinkResources true
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
             signingConfig signingConfigs.debug
+
+//            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
```
Comment (kjawadDeveloper2) [inline android/app/build.gradle]: Remove this dead code.

File: android/app/build.gradle [hunk 2]
```diff
@@ -77,4 +84,5 @@ flutter {
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation 'androidx.multidex:multidex:2.0.1'
+    implementation 'com.github.yalantis:ucrop:2.2.6'
```
Comment (kjawadDeveloper2) [inline android/app/build.gradle]: Remove this and then check if cropper is working or not in release mode.

File: i18n/ru-RU.json [hunk 3]
```diff
@@ -33,8 +33,22 @@
...
+  "royalty": "royalty",
+  "content_identifier": "доля личность",
+  "asset_uri": "актив URI ",
+  "copied_to_clipboard": "Скопировано в буфер обмена",
+  "save": "сберечь",
+  "continue": "Продолжать",
+  "read_more": "читать далее",
+  "collapse": "Крах",
+  "picking_file_error": "Ошибка при выборе файла",
+  "compressing_file_failure": "Ошибка при сжатии файла"
```
Comment (kjawadDeveloper2) [inline i18n/ru-RU.json]: url_launching_error is not here.

File: lib/easel_provider.dart [hunk 4]
```diff
@@ -239,7 +246,7 @@ class EaselProvider extends ChangeNotifier {
   }
 
   Future<void> delayLoading() async {
-    Future.delayed(const Duration(seconds: 3));
+    await Future.delayed(const Duration(seconds: 2));
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: Why we are delaying loading?

File: lib/easel_provider.dart [hunk 5]
```diff
@@ -950,6 +972,10 @@ class EaselProvider extends ChangeNotifier {
   Future<void> deleteNft(int? id) async {
     await repository.deleteNft(id!);
   }
+
+  void toHashtagList(String hashtag) {
+    hashtagsList = hashtag.split("#");
```
Comment (kjawadDeveloper2) [inline lib/easel_provider.dart]: Add constant for hash tag.

File: lib/repository/repository.dart [hunk 6]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function picks a file with the given format from device storage
+  /// Input: [format] it is the file format which needs to be picked from local storage
+  /// returns [PickedFileModel] the selected file or [Failure] if aborted
+  Future<Either<Failure, PickedFileModel>> pickFile(NftFormat format);
+
+  /// This function takes the file and returns a compressed version of that file
+  /// Input: [file] it takes the file that needs to be compressed
+  /// Output: [File] returns the compressed file
+  Future<Either<Failure, File?>> compressAndGetFile(File file);
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart]: It should be a file if there is no file throw an error. 

Output documentation is not according to our standards.

File: lib/repository/repository.dart [hunk 7]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function checks if a file path extension svg or not
+  /// Input: [filePath] the path of selected file
+  /// Output: [True] if the filepath has svg extension and [False] otherwise
+  bool isSvgFile(String filePath);
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart]: isSvgFile should be in extension.

File: lib/repository/repository.dart [hunk 8]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function checks if a file path extension svg or not
+  /// Input: [filePath] the path of selected file
+  /// Output: [True] if the filepath has svg extension and [False] otherwise
+  String getExtension(String fileName);
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart:132]: This should also be in extension.

File: lib/repository/repository.dart [hunk 9]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function is used to get the file size in GBs
+  /// Input: [fileLength] the file length in bytes
+  /// Output: [double] returns the file size in GBs in double format
+  double getFileSizeInGB(int fileLength);
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart:137]: This should also be an extension.

File: lib/repository/repository.dart [hunk 10]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function is used to get the file size in String format
+  /// Input: [fileLength] the file length in bytes and [precision] sets to [2] if not given
+  /// Output: [String] returns the file size in String format
+  String getFileSizeString({required int fileLength, int precision = 2});
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart:142]: This should also be in extension.

File: lib/repository/repository.dart [hunk 11]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function is used to generate the NFT link to be shared with others after publishing
+  /// Input: [recipeId] and [cookbookId] used in the link generation as query parameters
+  /// Output: [String] returns the generated NFTs link to be shared with others
+  String generateEaselLinkForShare({required String recipeId, required String cookbookId});
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart:147]: This should also be in extension.

File: lib/repository/repository.dart [hunk 12]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function is used to generate the NFT link to be open in the pylons wallet
+  /// Input: [recipeId] and [cookbookId] used in the link generation as query parameters
+  /// Output: [String] returns the generated NFTs link to be shared with others
+  String generateEaselLinkForOpeningInPylonsApp({required String recipeId, required String cookbookId});
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart]: This should also be an extension.

File: lib/repository/repository.dart [hunk 13]
```diff
@@ -112,14 +115,59 @@ abstract class Repository {
...
+
+  /// This function is used to launch the link generated and open the link in external source platform
+  /// Input: [url] is the link to be launched by the launcher
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart:150]: Output doc is not there.

File: lib/repository/repository.dart [hunk 14]
```diff
@@ -275,11 +323,74 @@ class RepositoryImp implements Repository {
       NFT? data = await localDataSource.getNft(id);
       if (data == null) {
         return Left(CacheFailure("something_wrong".tr()));
-      }
+      } else {
```
Comment (kjawadDeveloper2) [inline lib/repository/repository.dart]: We dont need to put else there.

File: lib/screens/creator_hub/widgets/delete_confirmation_dialog.dart [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+import 'package:easel_flutter/utils/constants.dart';
+import 'package:easel_flutter/utils/dependency_injection/dependency_injection_container.dart';
+import 'package:easel_flutter/utils/easel_app_theme.dart';
+import 'package:easel_flutter/widgets/clipped_button.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+TextStyle kDeleteHeaderTextStyle = TextStyle(fontSize: 14.sp, fontFamily: 'UniversalSans', color: EaselAppTheme.kWhite, fontWeight: FontWeight.w600);
```
Comment (kjawadDeveloper2) [inline lib/screens/creator_hub/widgets/delete_confirmation_dialog.dart]: Get this from constant.

File: lib/screens/describe_screen.dart [hunk 16]
```diff
@@ -165,25 +163,47 @@ class _DescribeScreenState extends State<DescribeScreen> {
...
+                        btnText: "save".tr(),
+                        showArrow: false,
+                        isBlue: false,
+                        isRed: false,
+                        mobileScreenButtonWidth: 0.4,
+                      ),
+                      PylonsButton(
+                        onPressed: () async {
+                          FocusScope.of(context).unfocus();
+                          if (_formKey.currentState!.validate()) {
```
Comment (kjawadDeveloper2) [inline lib/screens/describe_screen.dart]: Handle negative cases to avoid nested if.

File: lib/screens/describe_screen.dart [hunk 17]
```diff
@@ -165,25 +163,47 @@ class _DescribeScreenState extends State<DescribeScreen> {
...
-                        if (_formKey.currentState!.validate()) {
-                          if (_artNameFieldError.isEmpty && _artistNameFieldError.isEmpty && _descriptionFieldError.isEmpty) {
-                            context.read<EaselProvider>().updateNftFromDescription(provider.nft.id!);
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      PylonsButton(
+                        onPressed: () async {
+                          FocusScope.of(context).unfocus();
+                          if (_formKey.currentState!.validate()) {
```
Comment (kjawadDeveloper2) [inline lib/screens/describe_screen.dart]: Handle negative cases to avoid nested if.

File: lib/screens/preview_screen.dart [hunk 18]
```diff
@@ -59,8 +66,16 @@ class _PreviewScreenState extends State<PreviewScreen> {
               child: Align(
                 alignment: Alignment.bottomRight,
                 child: PylonsButton(
-                    onPressed: () {
-                      onUploadPressed();
+                    onPressed: () async {
+                      final result = await onUploadPressed();
+
+                      if (result) {
```
Comment (kjawadDeveloper2) [inline lib/screens/preview_screen.dart]: if something goes wrong where are we showing the error?

File: lib/screens/price_screen.dart [hunk 19]
```diff
@@ -233,22 +237,41 @@ class _PriceScreenState extends State<PriceScreen> {
...
-                        if (checkTextFields()) {
-                          context.read<EaselProvider>().updateNftFromPrice(nft!.id!);
-                          context.read<HomeViewModel>().pageController.nextPage(duration: const Duration(milliseconds: kPageAnimationTimeInMillis), curve: Curves.easeIn);
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    PylonsButton(
+                      onPressed: () async {
+                        FocusScope.of(context).unfocus();
+                        if (_formKey.currentState!.validate()) {
```
Comment (kjawadDeveloper2) [inline lib/screens/price_screen.dart]: Handle negative cases.

File: lib/screens/price_screen.dart [hunk 20]
```diff
@@ -233,22 +237,41 @@ class _PriceScreenState extends State<PriceScreen> {
...
+                      showArrow: false,
+                      isBlue: false,
+                      isRed: false,
+                      mobileScreenButtonWidth: 0.4,
+                    ),
+                    PylonsButton(
+                      onPressed: () async {
+                        FocusScope.of(context).unfocus();
+                        if (_formKey.currentState!.validate()) {
+                          if (checkTextFields()) {
```
Comment (kjawadDeveloper2) [inline lib/screens/price_screen.dart]: Handle negative cases first.

File: lib/screens/published_screen.dart [hunk 21]
```diff
@@ -1,10 +1,547 @@
...
+  State<PublishedScreen> createState() => _PublishedScreenState();
+}
+
+class _PublishedScreenState extends State<PublishedScreen> {
+  @override
+  void initState() {
+    super.initState();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      Navigator.of(context).push(
```
Comment (kjawadDeveloper2) [inline lib/screens/published_screen.dart]: Why we are pushing a new screen?

File: lib/screens/published_screen.dart [hunk 22]
```diff
@@ -1,10 +1,547 @@
...
+  State<PublishedNewScreen> createState() => _PublishedNewScreenState();
+}
+
+class _PublishedNewScreenState extends State<PublishedNewScreen> {
+  var repository = GetIt.I.get<Repository>();
+  var easelProvider = GetIt.I.get<EaselProvider>();
+  var homeViewModel = GetIt.I.get<HomeViewModel>();
+
+  @override
+  initState() {
```
Comment (kjawadDeveloper2) [inline lib/screens/published_screen.dart:51]: Add void here.

File: lib/screens/published_screen.dart [hunk 23]
```diff
@@ -1,10 +1,547 @@
...
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final viewModel = context.watch<EaselProvider>();
+
+    return Consumer<EaselProvider>(builder: (_, easelProvider, __) {
```
Comment (kjawadDeveloper2) [inline lib/screens/published_screen.dart]: If we had used watch there is no need for consumer.

File: lib/screens/published_screen.dart [hunk 24]
```diff
@@ -1,10 +1,547 @@
...
+                      owner: widget.nft.type == NftType.TYPE_RECIPE.name ? "you".tr() : widget.nft.creator,
+                    ),
+                    const SizedBox(
+                      height: 20,
+                    ),
+                    easelProvider.nft.assetType == kVideoText ? const VideoProgressWidget(darkMode: true, isForFile: false) : const SizedBox(),
+                  ],
+                ),
+              )
+            ] else ...[
```
Comment (kjawadDeveloper2) [inline lib/screens/published_screen.dart]: No need for else put a good if condition.

File: lib/screens/published_screen.dart [hunk 25]
```diff
@@ -1,10 +1,547 @@
...
+                                    children: [
+                                      buildRow(
+                                        title: "currency".tr(),
+                                        subtitle: widget.nft.isFreeDrop ? kPylonText : (viewModel.supportedDenomList.firstWhere((denom) => denom.symbol == widget.nft.denom).name),
+                                      ),
+                                      SizedBox(height: 2.h),
+                                      buildRow(
+                                        title: "price".tr(),
+                                        subtitle: widget.nft.isFreeDrop
+                                            ? "0"
```
Comment (kjawadDeveloper2) [inline lib/screens/published_screen.dart]: avoid nested ternary operations use a method.

File: lib/screens/published_screen.dart [hunk 26]
```diff
@@ -1,10 +1,547 @@
...
+                                              title: "publish".tr(),
+                                              bgColor: EaselAppTheme.kLightRed,
+                                              textColor: EaselAppTheme.kWhite,
+                                              onPressed: () async {
+                                                bool isRecipeCreated = await easelProvider.verifyPylonsAndMint(nft: easelProvider.nft);
+                                                if (!isRecipeCreated) {
+                                                  return;
+                                                }
+                                                easelProvider.disposeAudioController();
+                                                Navigator.of(context).pushNamedAndRemoveUntil(RouteUtil.kRouteCreatorHub, (route) => false);
```
Comment (kjawadDeveloper2) [inline lib/screens/published_screen.dart]: if the nft is audio there is no need for disposing an audio controller.

File: lib/utils/file_utils_helper.dart [hunk 27]
```diff
@@ -15,21 +19,21 @@ abstract class FileUtilsHelper {
...
   /// This function takes the file and returns a compressed version of that file
   /// Input: [file] it takes the file that needs to be compressed
   /// Output: [File] returns the compressed file
   Future<File?> compressAndGetFile(File file);
 
   /// This function checks if a file path extension svg or not
-  /// Input: [filePath] the file path
-  /// Output: [true] if the filepath has svg extension and [false] otherwise
+  /// Input: [filePath] the path of selected file
+  /// Output: [True] if the filepath has svg extension and [False] otherwise
```
Comment (kjawadDeveloper2) [inline lib/utils/file_utils_helper.dart]: Move it to extension.

File: lib/utils/file_utils_helper.dart [hunk 28]
```diff
@@ -15,21 +19,21 @@ abstract class FileUtilsHelper {
   /// This function picks a file with the given format from device storage
   /// Input: [format] it is the file format which needs to be picked from local storage
   /// returns [PlatformFile] the selected file or null if aborted
```
Comment (kjawadDeveloper2) [inline lib/utils/file_utils_helper.dart:18]: Update audio files .

File: lib/utils/file_utils_helper.dart [hunk 29]
```diff
@@ -38,30 +42,40 @@ abstract class FileUtilsHelper {
   double getFileSizeInGB(int fileLength);
```
Comment (kjawadDeveloper2) [inline lib/utils/file_utils_helper.dart]: Move it to extension.

File: lib/utils/file_utils_helper.dart [hunk 30]
```diff
@@ -139,4 +169,34 @@ class FileUtilsHelperImpl implements FileUtilsHelper {
...
+
+  Future<String> cropImage({required String filePath}) async {
+    try {
+      CroppedFile? croppedFile = await imageCropper.cropImage(
+        sourcePath: filePath,
+        aspectRatioPresets: [CropAspectRatioPreset.square, CropAspectRatioPreset.ratio3x2, CropAspectRatioPreset.original, CropAspectRatioPreset.ratio4x3, CropAspectRatioPreset.ratio16x9],
+        uiSettings: [
+          AndroidUiSettings(toolbarTitle: 'Pylons', toolbarColor: EaselAppTheme.kBlue, toolbarWidgetColor: Colors.white, initAspectRatio: CropAspectRatioPreset.original, lockAspectRatio: false),
+          IOSUiSettings(
+            title: 'Pylons',
```
Comment (kjawadDeveloper2) [inline lib/utils/file_utils_helper.dart]: Pylons should be a constant.

File: lib/widgets/audio_widget.dart [hunk 31]
```diff
@@ -219,13 +188,16 @@ class _AudioWidgetState extends State<AudioWidget> with WidgetsBindingObserver {
...
   void audioThumbnailPicker() async {
-    final result = await FileUtils.pickFile(NftFormat.supportedFormats[0]);
-    if (result != null) {
-      final loading = Loading()..showLoading(message: "compressing_thumbnail".tr());
-      final file = await FileUtils.compressAndGetFile(File(result.path!));
-      easelProvider.setAudioThumbnail(file);
-      loading.dismiss();
+    final pickedFile = await repository.pickFile(NftFormat.supportedFormats[0]);
+    final result = pickedFile.getOrElse(() => PickedFileModel(path: "", fileName: "", extension: ""));
+    if (result.path == "") {
```
Comment (kjawadDeveloper2) [inline lib/widgets/audio_widget.dart]: use .isEmpty

File: lib/widgets/video_progress_widget.dart [hunk 32]
```diff
@@ -35,88 +33,84 @@ class VideoProgressWidget extends StatelessWidget {
...
-                  SizedBox(
-                    height: 15.h,
-                    child: Row(
-                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        padding: EdgeInsets.only(right: 30.w, bottom: 10.h, top: 10.h, left: 25.w),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            if (easelProvider.videoLoadingError.isNotEmpty)
+              const SizedBox()
```
Comment (kjawadDeveloper2) [inline lib/widgets/video_progress_widget.dart:42]: Why we are not showing the video loading error to use. 

there is no need for showing empty sized boxes.

File: lib/widgets/video_progress_widget.dart [hunk 33]
```diff
@@ -35,88 +33,84 @@ class VideoProgressWidget extends StatelessWidget {
...
+            else
+              easelProvider.isVideoLoading
+                  ? const SizedBox()
```
Comment (kjawadDeveloper2) [inline lib/widgets/video_progress_widget.dart:45]: Show progress indicator. if its loading.

File: lib/widgets/video_progress_widget.dart [hunk 34]
```diff
@@ -35,88 +33,84 @@ class VideoProgressWidget extends StatelessWidget {
...
+                              bufferedColor: darkMode ? EaselAppTheme.kBlack.withOpacity(0.7) : EaselAppTheme.kWhite.withOpacity(0.7),
+                            ),
+                          ),
                         ),
                       ],
                     ),
-                  ),
-              ],
-            ),
+            if (easelProvider.videoLoadingError.isNotEmpty || easelProvider.isVideoLoading)
```
Comment (kjawadDeveloper2) [inline lib/widgets/video_progress_widget.dart:82]: Change if statements . Dont use it for empty Sized boxes.

File: lib/widgets/video_progress_widget.dart [hunk 35]
```diff
@@ -35,88 +33,84 @@ class VideoProgressWidget extends StatelessWidget {
...
+                          if (snapshot.hasData) {
+                            final String duration = _getDuration(snapshot.data!);
+                            return Padding(
+                              padding: EdgeInsets.only(left: 8.w),
+                              child: Text(
+                                duration,
+                                style: TextStyle(color: darkMode ? EaselAppTheme.kWhite : EaselAppTheme.kBlack),
+                              ),
+                            );
+                          } else {
```
Comment (kjawadDeveloper2) [inline lib/widgets/video_progress_widget.dart]: Avoid else from here.

File: test/utils/file_utils_test.dart [hunk 36]
```diff
@@ -1,15 +1,17 @@
...
-    FileUtilsHelperImpl fileUtilsHelperImpl = FileUtilsHelperImpl();
-    String recipeId = "testRecipe";
-    String cookbookId = "cookbookId";
 
-    String expectedDynamicLink = 'https://wallet.pylons.tech/?action=purchase_nft&recipe_id=$recipeId&cookbook_id=$cookbookId';
 
-    var dynamicLink = fileUtilsHelperImpl.generateEaselLink(recipeId: recipeId, cookbookId: cookbookId);
-    expect(expectedDynamicLink, dynamicLink);
-  });
+void main() {
```
Comment (kjawadDeveloper2) [inline test/utils/file_utils_test.dart:6]: Activate this test again.

File: android/app/build.gradle (hunk 2 above)
Comment (faizhashmiRnS) [reply android/app/build.gradle]: It is working properly in release mode without this dependency.

File: i18n/ru-RU.json (hunk 3 above)
Comment (faizhashmiRnS) [reply i18n/ru-RU.json]: Added.

File: pubspec.yaml [hunk 37]
```diff
@@ -75,15 +75,18 @@ dependencies:
...
   just_audio: 0.9.24
   audio_video_progress_bar: ^0.10.0
-  model_viewer_plus: 1.1.2
+  model_viewer_plus: 1.3.0
 
   dartz: ^0.10.0
 
   internet_connection_checker: ^0.0.1+4
 
+  detectable_text_field: ^2.0.3
```
Comment (kjawadDeveloper2) [inline pubspec.yaml]: Remove ^ this from the library.

File: lib/widgets/video_widget.dart [hunk 38]
```diff
@@ -133,62 +127,74 @@ class _VideoWidgetState extends State<VideoWidget> {
...
+                ],
                 SizedBox(
                   height: 10.w,
                 ),
-                Padding(
-                  padding: EdgeInsets.only(bottom: 70.h),
-                  child: VideoProgressWidget(darkMode: false, isForFile: widget.isForFile),
-                ),
-                if (shouldShowThumbnailButtonOrStepsOrNot()) ...[
+                if (shouldShowThumbnailButton()) ...[
```
Comment (kjawadDeveloper2) [inline lib/widgets/video_widget.dart]: Dont use spread operator for only one child.

File: lib/widgets/video_progress_widget.dart [hunk 39]
```diff
@@ -35,88 +33,97 @@ class VideoProgressWidget extends StatelessWidget {
...
+                    ),
+                  ],
+                ),
+              ),
+          ],
         ));
   }
 }
+
+class BuildActions extends StatelessWidget {
```
Comment (kjawadDeveloper2) [inline lib/widgets/video_progress_widget.dart]: Why its unused?

File: lib/widgets/pylons_button.dart [hunk 40]
```diff
@@ -1,58 +1,81 @@
...
       onTap: () {
         onPressed();
       },
+      child: ClipPath(
+        clipper: PylonsButtonClipper(),
+        child: Container(
+          width: isTablet ? 0.3.sw : mobileScreenButtonWidth.sw,
+          height: isTablet ? 0.07.sw : 0.12.sw,
+          decoration: BoxDecoration(
+            color: isBlue
```
Comment (kjawadDeveloper2) [inline lib/widgets/pylons_button.dart]: The way we are handling color is not acceptable.

File: lib/widgets/audio_widget.dart [hunk 41]
```diff
@@ -63,125 +66,91 @@ class _AudioWidgetState extends State<AudioWidget> with WidgetsBindingObserver {
...
-                  if (shouldShowThumbnailButtonOrStepsOrNot()) ...[
-                    _buildThumbnailButton(),
-                  ],
+                          ],
+                        ))
+                    : const SizedBox(),
+                SizedBox(
+                  height: 120.0.h,
+                ),
+                if (shouldShowThumbnailButtonOrStepsOrNot()) ...[
```
Comment (kjawadDeveloper2) [inline lib/widgets/audio_widget.dart]: spread operator should not be use for a single child.

File: lib/utils/read_more.dart [hunk 42]
```diff
@@ -0,0 +1,239 @@
...
+
+        return RichText(
+          textAlign: textAlign,
+          textDirection: textDirection,
+          textScaleFactor: textScaleFactor,
+          text: textSpan,
+        );
+      },
+    );
+    if (widget.semanticsLabel != null) {
```
Comment (kjawadDeveloper2) [inline lib/utils/read_more.dart]: i dont think we are using sematic labels.

File: lib/utils/file_utils_helper.dart [hunk 43]
```diff
@@ -139,4 +153,34 @@ class FileUtilsHelperImpl implements FileUtilsHelper {
...
+      print(e);
+      return "";
+    }
+  }
+
+  @override
+  String generateEaselLinkForOpeningInPylonsApp({required String recipeId, required String cookbookId}) {
+    return Uri.https('pylons.page.link', "/", {
+      "amv": "1",
+      "apn": "tech.pylons.wallet",
```
Comment (kjawadDeveloper2) [inline lib/utils/file_utils_helper.dart]: make apn and ibi constants.

File: lib/utils/file_utils_helper.dart [hunk 44]
```diff
@@ -15,21 +17,16 @@ abstract class FileUtilsHelper {
...
-  /// Output: [true] if the filepath has svg extension and [false] otherwise
+  /// Input: [filePath] the path of selected file
+  /// Output: [True] if the filepath has svg extension and [False] otherwise
   bool isSvgFile(String filePath);
 
   /// This function checks if a file path extension svg or not
-  /// Input: [filePath] the path of the file
-  /// Output: [true] if the filepath has svg extension and [false] otherwise
+  /// Input: [filePath] the path of selected file
+  /// Output: [True] if the filepath has svg extension and [False] otherwise
```
Comment (kjawadDeveloper2) [inline lib/utils/file_utils_helper.dart]: This should be in extension.

File: lib/utils/extension_util.dart [hunk 45]
```diff
@@ -112,3 +114,41 @@ extension MyStringSnackBar on String {
...
+    return "$kWalletWebLink/?action=purchase_nft&recipe_id=$this&cookbook_id=$cookbookId";
+  }
+}
+
+
+extension GenerateEaselLinkToOpenPylons on String {
+  String generateEaselLinkForOpeningInPylonsApp({required String cookbookId}) {
+    return Uri.https('pylons.page.link', "/", {
+      "amv": "1",
+      "apn": "tech.pylons.wallet",
```
Comment (kjawadDeveloper2) [inline lib/utils/extension_util.dart:148]: Make these constants.

File: lib/screens/published_screen.dart [hunk 46]
```diff
@@ -1,10 +1,672 @@
...
+                      ),
+                    ),
+                    _title(
+                      nft: widget.nft,
+                      owner: widget.nft.type == NftType.TYPE_RECIPE.name ? "you".tr() : widget.nft.creator,
+                    ),
+                    const SizedBox(
+                      height: 20,
+                    ),
+                    viewModel.nft.assetType == kVideoText ? const VideoProgressWidget(darkMode: true, isForFile: false) : const SizedBox(),
```
Comment (kjawadDeveloper2) [inline lib/screens/published_screen.dart]: Use widget builder here.

File: android/app/build.gradle [hunk 47]
```diff
@@ -53,19 +53,19 @@ android {
     }
 
     signingConfigs {
-//        debug  {
```
Comment (kjawadDeveloper2) [inline android/app/build.gradle]: Revert this because i need them when i upload a build.

File: i18n/ru-RU.json [hunk 48]
```diff
@@ -36,5 +36,29 @@
...
+  "asset_uri": "актив URI ",
+  "copied_to_clipboard": "Скопировано в буфер обмена",
+  "save": "сберечь",
+  "continue": "Продолжать",
+  "read_more": "читать далее",
+  "collapse": "Крах",
+  "picking_file_error": "Ошибка при выборе файла",
+  "compressing_file_failure": "Ошибка при сжатии файла",
+  "url_launching_error": "Ошибка при запуске URL",
+  "size_error": "Size Error",
```
Comment (mustafapylons) [inline i18n/ru-RU.json]: Are these not supposed to be translated to Russian?

**Review by mustafapylons — COMMENTED** [review]
One issue with translation, otherwise code is good.

**Review by mustafapylons — APPROVED** [review]
Looks good to me now.

File: lib/screens/tutorial_screen.dart [hunk 49]
```diff
@@ -132,6 +87,25 @@ class _TutorialScreenState extends State<TutorialScreen> {
...
+            padding: EdgeInsets.symmetric(horizontal: 0.22.sw),
+            child: Image.asset(
+              item['image'],
+              height: isTablet ? 140.w : 200.w,
+              width: isTablet ? 140.w : 200.w,
+              fit: BoxFit.fill,
+            )),
+        SizedBox(height: 0.15.sh),
+        Text(item['header'], style: TextStyle(fontSize: isTablet ? 16.sp : 18.sp, fontWeight: FontWeight.w800, color: EaselAppTheme.kDartGrey), textAlign: TextAlign.center),
+        const SizedBox(height: 15),
```
Comment (Amean) [inline lib/screens/tutorial_screen.dart:104]: Should be .h for SizedBox?

**Review by Amean — APPROVED** [review]

**Pull request merged**